tls/s2n_connection.c

```diff
@@ -1356,7 +1359,12 @@ int s2n_connection_get_peer_cert_chain(const struct s2n_connection *conn, struct
             s2n_openssl_x509_stack_pop_free);
     POSIX_ENSURE_REF(cert_chain_validated);
 
-    for (size_t cert_idx = 0; cert_idx < sk_X509_num(cert_chain_validated); cert_idx++) {
+    int cert_count = sk_X509_num(cert_chain_validated);
+
+    /* Check to ensure cert_count is non-negative before casting */
+    POSIX_ENSURE_GTE(cert_count, 0);
```

Comment:
nit: I'm not really sure if these comments add very much. If the check comes right after the call, it might be clear enough what the check is for.
```suggestion
    POSIX_ENSURE_GTE(cert_count, 0);
```